src/main/java/com/aws/greengrass/lifecyclemanager/Kernel.java

```diff
@@ -166,6 +166,33 @@ public Kernel() {
         context.put(SERVICE_TYPE_TO_CLASS_MAP_KEY, typeToClassMap);
     }
 
+    /**
+     * Construct the Kernel to be used for connectivity validation.
+     *
+     * @param currentConfig         current device configs
+     * @param deploymentConfig      Map of the configs being merged in from deployment
+     * @param deploymentTimestamp   timestamp of the deployment being merged in
+     */
+    public Kernel(DeviceConfiguration currentConfig, Map<String, Object> deploymentConfig,
+                  long deploymentTimestamp) {
+        context = new Context();
+        config = new Configuration(context);
+        // Copy the current device configuration Map to preserve time stamps
+        config.copyFrom(currentConfig.getConfig());
+        // Attempt to merge the deployment configs using the deployment time stamp
+        config.mergeMap(deploymentTimestamp, deploymentConfig);
+        try {
+            config.waitConfigUpdateComplete();
+        } catch (InterruptedException e) {
+            logger.atError().log("Wait for config update complete was interrupted");
```

Comment:
Handle the interruption correctly. Re-interrupt the thread.

`Thread.currentThread().interrupt()`.